### PR TITLE
Resolving ROI issue & resolving bug in fused_crop_decoder

### PR DESCRIPTION
**Pull request by swetha097**
This PR 
1. Resolves the ROI issue 
2. Bug in the fused crop decoder when the x & x_diff coordinates are not same

### CODE REVIEW

File: rocAL/rocAL/source/decoders/image/fused_crop_decoder.cpp [hunk 1]
```diff
@@ -138,8 +138,10 @@ Decoder::Status FusedCropTJDecoder::decode(unsigned char *input_buffer, size_t i
...
     }
-    
-   // std::cout<<"Fused Crop Decoder <x,y, w, h>: " << x1 << " " << y1 << " " << crop_width << " " << crop_height << std::endl;
+    if(crop_width > max_decoded_width)
+        crop_width = max_decoded_width;
+    if(crop_height > max_decoded_height)
+        crop_height = max_decoded_height;
     //TODO : Turbo Jpeg supports multiple color packing and color formats, add more as an option to the API TJPF_RGB, TJPF_BGR, TJPF_RGBX, TJPF_BGRX, TJPF_RGBA, TJPF_GRAY, TJPF_CMYK , ...
     if( tjDecompress2_partial(m_jpegDecompressor,
                       input_buffer,
```
Comment (sampath1117) [inline rocAL/rocAL/source/decoders/image/fused_crop_decoder.cpp]: Align parameters

File: rocAL/rocAL/source/decoders/image/fused_crop_decoder.cpp [hunk 2]
```diff
@@ -158,20 +160,16 @@ Decoder::Status FusedCropTJDecoder::decode(unsigned char *input_buffer, size_t i
     }
 
     if (x1 != x1_diff) {
-        //std::cout << "x_off changed by tjpeg decoder " << x1 << " " << x1_diff << std::endl;
         unsigned char *src_ptr_temp, *dst_ptr_temp;
-        unsigned int elements_in_row = max_decoded_width * planes;
         unsigned int elements_in_crop_row = crop_width * planes;
-        //unsigned int remainingElements =  elements_in_row - elements_in_crop_row;
         unsigned int xoffs = (x1-x1_diff) * planes;   // in case x1 gets adjusted by tjpeg decoder
```
Comment (sampath1117) [inline rocAL/rocAL/source/decoders/image/fused_crop_decoder.cpp]: Space before and after - operator in Line 165

File: rocAL/rocAL/source/decoders/image/fused_crop_decoder.cpp [hunk 3]
```diff
@@ -138,8 +138,10 @@ Decoder::Status FusedCropTJDecoder::decode(unsigned char *input_buffer, size_t i
...
         }
     }
-    
-   // std::cout<<"Fused Crop Decoder <x,y, w, h>: " << x1 << " " << y1 << " " << crop_width << " " << crop_height << std::endl;
+    if(crop_width > max_decoded_width)
+        crop_width = max_decoded_width;
+    if(crop_height > max_decoded_height)
+        crop_height = max_decoded_height;
     //TODO : Turbo Jpeg supports multiple color packing and color formats, add more as an option to the API TJPF_RGB, TJPF_BGR, TJPF_RGBX, TJPF_BGRX, TJPF_RGBA, TJPF_GRAY, TJPF_CMYK , ...
     if( tjDecompress2_partial(m_jpegDecompressor,
```
Comment (sampath1117) [inline rocAL/rocAL/source/decoders/image/fused_crop_decoder.cpp]: Remove space before tjDecompress2_partial()

**Comment by shobana-mcw** [comment]
@swetha097 Resolve the comments from Sampath.

File: rocAL/rocAL/source/decoders/image/fused_crop_decoder.cpp [hunk 4]
```diff
@@ -138,40 +138,38 @@ Decoder::Status FusedCropTJDecoder::decode(unsigned char *input_buffer, size_t i
             y1 =  (original_image_height - crop_height) / 2;
         }
     }
-    
-   // std::cout<<"Fused Crop Decoder <x,y, w, h>: " << x1 << " " << y1 << " " << crop_width << " " << crop_height << std::endl;
+    if(crop_width > max_decoded_width)
+        crop_width = max_decoded_width;
+    if(crop_height > max_decoded_height)
+        crop_height = max_decoded_height;
```
Comment (shobana-mcw) [inline rocAL/rocAL/source/decoders/image/fused_crop_decoder.cpp:145]: Add a comment why we are handling this case.

**Comment by shobana-mcw** [comment]
@swetha097 Pls resolve this comment and issue the PR.

File: rocAL/rocAL/source/decoders/image/fused_crop_decoder.cpp (hunk 4 above)
Comment (swetha097) [reply rocAL/rocAL/source/decoders/image/fused_crop_decoder.cpp:145]: Done

**Pull request closed without merging**